rust/libnewsboat-ffi/src/utils.rs

```diff
@@ -9,6 +9,13 @@ use std::ffi::{CStr, CString};
 use std::path;
 use std::ptr;
 
+#[cxx::bridge(namespace = "newsboat::utils")]
+mod ffi {
```

Comment:
Is there a reason to call this `mod ffi` while calling it `mod bridged` in other locations?

Reply:
Yeah, I'm trying to highlight that these mods contain different "kinds" of functions.

`mod ffi` contains things that don't need any handwritten wrappers. When you look at bridges in `mod ffi`, you can assume that they refer directly to functions from libnewsboat crate, and also assume that on the C++ side these functions are called directly, with no additional wrappers.

`mod bridged` is for more complicated functions that need handwritten wrappers on one or both sides of the bridge. `utils::to_u` is a good example: on the C++ side, it's convenient to have the second argument of this function default to 0, so we have a wrapper there. When you look at bridges in `mod bridged`, you should check if there is a similarly-named function in the enclosing scope (i.e. a wrapper on the Rust side), and you should also check if C++ side wraps that function.

This distinction is also reflected in C++ namespaces that bridges use: `mod ffi` goes directly into the appropriate C++ namespace, like `newsboat::utils`. `mod bridged`, on the other hand, creates a sub-namespace, e.g. `newsboat::utils::bridged`, so you're less likely to call that bridge directly — forcing you to use a nice wrapper that's placed in `newsboat::utils`.

I feel like I don't explain this adequately enough. Does it even make sense? I guess these things should be written down in a little document somewhere, because they should apply to each and every bridge we write, and writing this in code comments is unsustainable.

Reply:
> I feel like I don't explain this adequately enough. Does it even make sense?

Makes perfect sense to me, thanks!

> I guess these things should be written down in a little document somewhere, because they should apply to each and every bridge we write, and writing this in code comments is unsustainable.

Placing explanations like this in an issue might be enough (e.g. as a comment on https://github.com/newsboat/newsboat/issues/286)

Reply:
An issue might not be a good place for this, because it will accumulate outdated info. I think I'll just link to this very comment, and stop doing that once we got enough bridges written that the convention becomes established. Sounds good?

Reply:
Sounds good to me